### PR TITLE
Add Grafana-managed LLM as an option

### PR DESCRIPTION
**Pull request by gerboland**
Adding Grafana-managed OpenAI as an option.

This is where the user has access to OpenAI via an account Grafana manages. This involves all requests being proxied via the llm-gateway, which implements rate limiting and adds the Grafana OpenAI key to requests.

A key requirement is that this option must have an explicit Opt-In action performed by the user before it is enabled. The history of the opt-in actions will (later) be saved for future reference.

Changes to the backend:
- extend Settings to include LLMGateway struct, which contains the URL to the llm-gateway (with fallback to env var), and an opt-in boolean. Both should be supplied by GCom.
- like in machine learning, also get a Token from GCom that contains the tenant_id and api key for the llm-gateway. Use this info to authenticate with the llm-gateway.
- grafanaOpenAIProxy is added which proxies requests to the llm-gateway.

Changes to the frontend:
- a redesign of the OpenAI settings page to include the Grafana-managed OpenAI as an option, along with User-supplied key option plus an additional Disabled option. [Visuals here](https://miro.com/app/board/uXjVNzkCgV0=/?share_link_id=406199692543).

Note: this PR does not include any logic to push updated settings to GCom.

### CODE REVIEW

**Review by yoziru — COMMENTED** [review]
General testing lgtm ✅ 

Added a few comments, apart from those:
- We should probably disable the OpenAI provider option in the Vector Embedder Provider when managed key is selected
- UI: it's easy to forget to hit save at the bottom here. Could we either auto-save on each change or pop up a modal if the user navigates away without saving?

File: src/components/AppConfig/AppConfig.tsx [hunk 1]
```diff
@@ -51,12 +59,51 @@ export const AppConfig = ({ plugin }: AppConfigProps) => {
...
+      return false;
+    }
+    return true;
+  };
+
+  const doSave = async () => {
+    if (!validateInputs()) {
+      return;
+    }
+    setIsUpdating(true);
```
[the remark this answers is not in the document]
Comment (gerboland) [reply src/components/AppConfig/AppConfig.tsx:75]: I wanted to leave that for a second PR. This was getting big.

File: src/components/AppConfig/AppConfig.tsx [hunk 2]
```diff
@@ -3,17 +3,25 @@ import { lastValueFrom } from 'rxjs';
...
-import { VectorConfig, VectorSettings } from './Vector';
 import { ShowHealthCheckResult } from './HealthCheck';
+import { LLMConfig } from './LLMConfig';
+import { OpenAISettings } from './OpenAI';
+import { VectorConfig, VectorSettings } from './Vector';
 
+///////////////////////
+export interface LLMGatewaySettings {
+  // Opt-in to LLMGateway?
+  optInStatus?: boolean;
```
[the remark this answers is not in the document]
Comment (gerboland) [reply src/components/AppConfig/AppConfig.tsx:18]: Yeah that's preferable, but I was trying to stay close to the key name in gcom though (llmOptInStatus).

File: src/components/AppConfig/LLMConfig.tsx [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+              secretsSet={secretsSet}
+              onChangeSecrets={onChangeSecrets}
+            />
+          </div>
+        )}
+
+        <Card isSelected={llmOption === 'disabled'} onClick={selectLLMDisabled} className={s.cardWithoutBottomMargin}>
+          <Card.Heading>Disable all LLM features in Grafana</Card.Heading>
+          <Card.Figure>
+            <Icon name="times" size="lg" />
```
[the remark this answers is not in the document]
Comment (gerboland) [reply src/components/AppConfig/LLMConfig.tsx:271]: Hmm, maybe, that's how the Card behaves... I wasn't planning on much UI tweaking until UX takes a look.

File: src/components/AppConfig/LLMConfig.tsx [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+                </>
+              )}
+            </Field>
+          </div>
+        )}
+        <Card isSelected={llmOption === 'openai'} onClick={selectOpenAI} className={s.cardWithoutBottomMargin}>
+          <Card.Heading>Use your own OpenAI account</Card.Heading>
+          <Card.Description>Enable LLM features in Grafana using your own OpenAI details</Card.Description>
+          <Card.Figure>
+            <Icon name="sitemap" size="lg" />
```
[the remark this answers is not in the document]
Comment (gerboland) [reply src/components/AppConfig/LLMConfig.tsx:252]: Yes I was just (a) lazy and (b) wanted to check their logo rights.

[the remark this answers is not in the document]
Comment (gerboland) [reply src/components/AppConfig/LLMConfig.tsx]: Done, though awkwardly. I couldn't find an elegant way to include svgs with the plugin packaging, so I created a react component including the raw svg

**Comment by gerboland** [comment]
CI passed but I think we've run out of github api requests, so drone cannot update the state here

**Review by yoziru — APPROVED** [review]
Screenshot 2024-02-02 at 16 41 04


Screenshot 2024-02-02 at 16 40 08

LGTM, sync on save (with gcom) can happen in separate PR

**Pull request merged**